### ISSUE TITLE
Bug: no-restricted-imports -- can't use slash in patterns[].group

### ISSUE THREAD

**Issue by ianfp** [eslint/eslint#16881]
### Environment

Node version: 16.15.0
~npm~ yarn version: 3.2.4
Local ESLint version: 8.33.0
Global ESLint version: none
Operating System: MacOS Ventura 13.0

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
"no-restricted-imports": [
    "error",
    {
        patterns: [
            {
                group: ["*/dist/*"],
                message:
                    "Imports from dist/ are not allowed. Perhaps the name you're trying to import is not exported?",
            },
        ],
    },
],
```
</details>

```js
import { SomeClass } from "@myorg/some-package/dist/esm/Interfaces";
```

### What did you expect to happen?

The import line above would be flagged as an error.

### What actually happened?

No error.

The import _does_ get flagged if my config omits the slashes:
```
"no-restricted-imports": [
    "error",
    {
        patterns: [
            {
                group: ["*dist*"],
                message:
                    "Imports from dist/ are not allowed. Perhaps the name you're trying to import is not exported?",
            },
        ],
    },
],
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @ianfp, thanks for the issue!

This works as intended, because patterns in the `no-restricted-imports` rule work per the [gitignore specification](https://git-scm.com/docs/gitignore).

> ```js
> group: ["*/dist/*"]
> ```

Since there's a slash in the middle, this matches relative to the root, so for example it matches `"foo/dist/baz"` but not `"foo/bar/dist/baz"`.

You would probably want to use `**` instead of `*`, to match at any level:

```js
group: ["**/dist/**"]
```

[ESLint Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiaW1wb3J0IHsgU29tZUNsYXNzIH0gZnJvbSBcIkBteW9yZy9zb21lLXBhY2thZ2UvZGlzdC9lc20vSW50ZXJmYWNlc1wiO1xuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJuby1yZXN0cmljdGVkLWltcG9ydHMiOlsiZXJyb3IiLHsicGF0dGVybnMiOlt7Imdyb3VwIjpbIioqL2Rpc3QvKioiXSwibWVzc2FnZSI6IkltcG9ydHMgZnJvbSBkaXN0LyBhcmUgbm90IGFsbG93ZWQuIFBlcmhhcHMgdGhlIG5hbWUgeW91J3JlIHRyeWluZyB0byBpbXBvcnQgaXMgbm90IGV4cG9ydGVkPyJ9XX1dfSwiZW52Ijp7ImVzNiI6dHJ1ZX19fQ==)
